### PR TITLE
some file system tables/magic love

### PR DESCRIPTION
**Pull request by poettering**
Inspired by this discussion:

https://github.com/systemd/systemd/pull/21333#pullrequestreview-805119204

### CODE REVIEW

File: src/basic/filesystems.c [hunk 1]
```diff
@@ -47,26 +57,50 @@ const FilesystemSet filesystem_sets[_FILESYSTEM_SET_MAX] = {
...
+                "rpc_pipefs\0"
                 "securityfs\0"
         },
         [FILESYSTEM_SET_COMMON_BLOCK] = {
                 .name = "@common-block",
                 .help = "Common block device filesystems",
                 .value =
                 "btrfs\0"
+                "erofs\0"
+                "exfat\0"
```
Comment (medhefgo) [inline src/basic/filesystems.c:99]: f2fs (😿), ntfs (non-fuse from recent new kernel)?

Comment (poettering) [reply src/basic/filesystems.c:99]: send a follow-up pr?

is f2fs that common? never heard of it?

did the new ntfs driver take over the old ntfs driver's magic and name?

Comment (medhefgo) [reply src/basic/filesystems.c:99]: > send a follow-up pr?

It's just a matter of hadding one measly line :(

You must've lived under a rock if you never heard of f2fs. It's the fs to use for flash drives and helps reducing wear. And unlike btrfs (which I only used briefly), it hasn't yet managed to get itself corrupted on my root fs.

(But yet, it probably depends on your definition of common and what this set is used for.)

Apparently, the new module's name is ntfs3 (and the old ntfs module nobody liked still exists). Is this the magic you're looking for (it is different from the old magic)?
https://github.com/torvalds/linux/blob/19901165d90fdca1e57c9baa0d5b4c63d15c476a/fs/ntfs3/super.c#L901

Comment (poettering) [reply src/basic/filesystems.c:99]: added.

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**